test/integration/bugs_test.go

```diff
@@ -160,148 +162,114 @@ func TestUnreachableHost(t *testing.T) {
 	t.Log(errDegraded)
 }
 
-func genLatestArchiveCheckPattern(prettyName string, check func(*testing.T, string, []string, *regexp.Regexp) error, pattern string) func(t *testing.T) {
+func genLatestArchiveCheckPattern(prettyName string, check archiveCheck, archive []string, patterns ...string) test {
 	return func(t *testing.T) {
-		err := latestArchiveCheckFiles(t, prettyName, check, pattern)
-		if err != nil {
-			t.Fatal(err)
+		if len(patterns) == 0 {
+			t.Fatal(prettyName, ": No patterns to check")
+		}
+		for _, pattern := range patterns {
+			err := checkArchiveFiles(t, prettyName, check, pattern, archive)
```

Comment:
Just minor - would it make sense to move the `checkArchiveFiles` (and probably also the `latestArchiveFiles`) function to `bugs_test.go`? 
I am not very experienced with Go tests and testing, but the `main_test.go` looks more like some test utils file (if it is so then we can leave these functions there) .

Reply:
I understand it as test utils file now too, checkArchiveFiles could make sense to move to bugs.go, I don't think so about latestArchiveFiles (I could want to import it to different test file).. But it would make sense to move all the util functions from main.go to different files, as it starts getting unnecessarily big, an it's not very readable what main.go is

Reply:
(should be done in different PR)

Reply:
I agree with @psimovec on this one. For now, it is fine, but it is starting to become a huge and not readable file.